packages/components/core/common-behaviors/error-state.ts

```diff
@@ -92,3 +103,43 @@ export class KbqErrorStateTracker implements CanUpdateErrorState {
         }
     }
 }
+
+/**
+ * Directive to automatically apply error-based styling to a host element
+ * when used within a form field component.
+ */
+@Directive({
+    standalone: true,
+    selector: '[kbqAutoColor]',
+    exportAs: 'kbqAutoColor',
+    host: {
+        '[class.kbq-error]': 'hasError'
+    }
+})
+export class KbqAutoColor implements AfterContentInit {
+    readonly autoColor = input(false, { alias: 'kbqAutoColor', transform: booleanAttribute });
+    protected readonly formField = inject(KBQ_FORM_FIELD_REF, { optional: true });
+    protected readonly cdr = inject(ChangeDetectorRef);
+    protected readonly destroyRef = inject(DestroyRef);
+
+    /** Flag that controls css-class on host when control state changes. */
+    hasError: boolean = false;
```

Comment:
меньше кода и не нужно будет управлять `cdr`
```suggestion
      /** @docs-private */
     readonly hasError = toSignal(this.formField?.control?.stateChanges.pipe(map(...))), { initialValue: false });
```